Feature: add a legend to leaflet-heatmap

LePew opened the pull request:
Hi. 
This script adds a Control on a leaflet-heatmap displaying min and max values + an img of the gradient.
Feel free to merge it with your repo :)
![heatmapcontrol](https://cloud.githubusercontent.com/assets/22963570/24008794/071f7990-0a73-11e7-8149-b25ec8c4b65c.jpg)

paalbra committed:
Fix broken paths in bower.json

pa7 committed:
Merge pull request #249 from paalbra/patch-1

Fix broken paths in bower.json

LePew committed:
Legend control for leaflet-heatmap

Inspired by https://www.patrick-wied.at/static/heatmapjs/example-legend-tooltip.html and https://github.com/dwilhelm89/LeafletSlider
Displays a legend box on your map showing min and max value of the gradient and a nifty gradient img.
Position legend as you wish like any standard Leaflet control (topleft, topright, ...) and customize with provided CSS file.

LePew committed:
CSS to tweak heatmapControl appearance

LePew committed:
Update heatmapControl.js

LePew committed:
Update heatmapControl.css

pa7 commented:
@LePew looks good in general, will add some minor feedback for changes to get into develop later today

pa7 commented on plugins/leaflet-heatmap/heatmapControl.css:
```diff
@@ -0,0 +1,26 @@
+    outline: 1px solid #000;
+    line-height: 1em;
+    font-size: 12px;
+    
```
nit: remove

pa7 commented on plugins/leaflet-heatmap/heatmapControl.js:
```diff
@@ -0,0 +1,103 @@
+            map.addControl(this);
+        }
+       //We need a redraw even if values didn't change, hence this little trick...
+        var data = this._extremeData; 
```
general nit: code needs consistent indentation

pa7 commented on plugins/leaflet-heatmap/heatmapControl.js:
```diff
@@ -0,0 +1,103 @@
+       //We need a redraw even if values didn't change, hence this little trick...
+        var data = this._extremeData; 
+	this._extremeData={min: undefined, max: undefined, gradient: {}};
+	console.log(data);
```
nit: remove console

pa7 commented on plugins/leaflet-heatmap/heatmapControl.js:
```diff
@@ -0,0 +1,103 @@
+	  //We could create the img once in 'onAdd' but someone may change the gradient config...
+	  var legendCtx = this._legendCanvas.getContext('2d');  
+	  var gradient = legendCtx.createLinearGradient(0, 0, 100, 1);
+	  var gradientImg = document.querySelector('#heatmapLegendGradient');
```
it would be better to use css classes for that and scope them to the control container (assuming L.DomUtil returns the dom node): 
```js
// ie
var gradientImg = this._heatmapControlContainer.querySelector('.heatmapLegendGradient');
```
This comes in handy if we have multiple instances of maps on one page (as the implementation right now would create duplicate ids.

**Same w/ the other ids (heatmapLegendMin, heatmapLegendMax)**. I know most of it is from example code, but it makes sense to improve the overall  quality before it gets into develop

LePew replied:
@pa7 You're right, I should've thought about it. Thanks for the feedback.

LePew committed:
Switch to class definition in CSS and scope to control container

Should be better if we have multiple heatmaps and controls displayed at the same time.
